velox/vector/DecodedVector.cpp

```diff
@@ -52,7 +52,8 @@ void DecodedVector::decode(
     bool loadLazy) {
   reset(end(vector.size(), rows));
   loadLazy_ = loadLazy;
-  if (loadLazy_ && (isLazyNotLoaded(vector) || vector.isLazy())) {
+  if ((vector.isLazy() && vector.asUnchecked<LazyVector>()->isLoaded()) ||
```

Comment:
This expression is a mouthful now; is there a way to make it more concise? Whether recursively there any lazy vectors not loaded?

Reply:
Replaced this with the following to make it easier to understand, lemme know what you think:

```
bool isTopLevelLazyAndLoaded =
      vector.isLazy() && vector.asUnchecked<LazyVector>()->isLoaded();
  if (isTopLevelLazyAndLoaded || (loadLazy_ && isLazyNotLoaded(vector))) {
    decode(*vector.loadedVector(), rows);
    return;
  }
```